myapp/spec/models/task_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'rails_helper'
+
+RSpec.describe 'Task', type: :model do
+  describe 'normal case' do
+    context 'all valid parameters' do
+      let(:task) {
+        Task.new(name: 'sample_task',
+                 description: 'hoge fuga',
+                 deadline_at: '2023-01-01T00:00')
+      }
+
+      it 'has no error' do
+        task.valid?
+        expect(task.errors.count).to eq 0
+      end
+    end
+  end
+
+  describe 'validation' do
+    describe 'name' do
+      context 'length is 0' do
+        let(:task) {
+          Task.new(name: '',
+                  description: 'hoge fuga',
+                  deadline_at: '2023-01-01T00:00')
+        }
+
+        it 'has validation error' do
+          task.valid?
+          expect(task.errors.count).to eq 1
+          expect(task.errors[:name].present?).to be true
+        end
+      end
```

Comment:
I prefer to write like below because I understand more easily.
But I hear there are some arguments for not doing too much, so as you like!

```ruby
describe 'name' do
  let(:task) {
    Task.new(name: name,
            description: 'hoge fuga',
            deadline_at: '2023-01-01T00:00')
  }

  context 'length is 0' do
    let(:name) { '' }

    it 'has no error' do
      task.valid?
      expect(task.errors.count).to eq 0
    end
  end

  context 'length is 255' do
    let(:name) { 'a' * 255 }

    it 'has no error' do
      task.valid?
      expect(task.errors.count).to eq 0
    end
  end
end
```

Reply:
That's looks also nice. Thank you for sharing it.